js/client/modules/@arangodb/testsuites/recovery_cluster.js

```diff
@@ -152,37 +155,53 @@ function runArangodRecovery (params) {
 
   if (params.setup) {
     params.args = Object.assign(params.args, additionalParams);
-    params.instanceInfo = pu.startInstance(params.options.protocol,
-                                           params.options,
-                                           params.args,
-                                           fs.join('recovery_cluster', params.count.toString()));
+    params.instanceManager = new im.instanceManager(params.options.protocol,
+                                                    params.options,
+                                                    params.args,
+                                                    fs.join('recovery_cluster',
+                                                            params.count.toString()));
+    params.instanceManager.prepareInstance();
+    params.instanceManager.launchTcpDump("");
+    if (!params.instanceManager.launchInstance()) {
+      return {
+        export: {
+          status: false,
+          message: 'failed to start server!'
+        }
+      };
+    }
   } else {
     print(BLUE + "Restarting cluster " + RESET);
-    pu.reStartInstance(params.options, params.instanceInfo, {});
+    params.instanceManager.reStartInstance();
     let tryCount = 10;
-    while(tryCount > 0 && !pu.checkServersGOOD(params.instanceInfo)) {
+    while(tryCount > 0 && !params.instanceManager._checkServersGOOD()) {
       print(RESET + "Waiting for all servers to go GOOD");
       internal.sleep(3); // give agency time to bootstrap DBServers
       --tryCount;
     }
   }
+  params.instanceManager.reconnect();
   let agentArgs = pu.makeArgs.arangosh(params.options);
-  agentArgs['server.endpoint'] = tu.findEndpoint(params.options, params.instanceInfo);
+  agentArgs['server.endpoint'] = params.instanceManager.findEndpoint();
   if (params.args['log.level']) {
     agentArgs['log.level'] = params.args['log.level'];
   }
-  agentArgs['temp.path'] = params.instanceInfo.rootDir;
+
   Object.assign(agentArgs, additionalTestParams);
-  require('internal').env.INSTANCEINFO = JSON.stringify(params.instanceInfo);
+  require('internal').env.INSTANCEINFO = JSON.stringify(params.instanceManager.getStructure());
   try {
-    pu.executeAndWait(pu.ARANGOSH_BIN, toArgv(agentArgs), params.options, 'arangosh', params.instanceInfo.rootDir, false);
+    pu.executeAndWait(pu.ARANGOSH_BIN, toArgv(agentArgs), params.options, 'arangosh', params.instanceManager.rootDir, false);
   } catch(err) {
     print('Error while launching test:' + err);
-    pu.shutdownInstance(params.instanceInfo, params.options, false);
+    params.instanceManager.shutdownInstance(false);
     return; // without test server test will for sure fail
   }
   if (params.setup) {
-    let dbServers = params.instanceInfo.arangods.slice().filter((a) => {return a.role === 'dbserver' || a.role === 'coordinator';});
+    let dbServers = params.instanceManager.arangods.slice().filter(
```

Comment:
Why slice()?

Reply:
let's see whether it works without - seems like cargo cult programming.